packages/reducers/app.ts

```diff
@@ -157,3 +161,35 @@ export const calculateCounterparties = (state) => () : number => {
 
   return Object.keys(friends).length - 1
 }
+
+export const calculateUcacBalances = (state) => (friendAddress: string) : Object => {
+  const recent = recentTransactions(state)
+  const user = getUser(state)()
+  const ucacBalances = {}
+
+  recent.map( tx => {
+    if(tx.creditorAddress === friendAddress) {
```

Comment:
This could be much simplified to the following (or something similar):
```
const multiplier = (tx.debtorAddress === friendAddress) ? 1 : -1
const value = multipler*tx.amount
const currency = getUcacCurrency(state)(tx.ucac)
if (ucacBalances[currency] === undefined)
	ucacBalances[currency] = value
else
	ucacBalances[currency] += value
```